runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -262,7 +342,13 @@ private void insertAllDependingFileDescriptorProtos(TableDescriptors.Builder tab
             }
 
             // Add the fileDescriptorProto into the tableDescriptor map.
-            tableDescriptorsBuilder.putFileDescriptors(fileDescriptorProto.getName(), fileDescriptorProto);
+            tableDescriptorsBuilder.putFileDescriptors(fileDescriptorProto.getName(),
+                    FileDescriptorProto.getDefaultInstance());
+
+            // Add the actual descriptor into a common pool of descriptors to avoid duplication
+            ProtobufFileName protoFileName = ProtobufFileName.newBuilder().setFileName(fileDescriptorProto.getName()).build();
+            ProtobufFileDescriptor protoFd = ProtobufFileDescriptor.newBuilder().setFileDescriptor(fileDescriptorProto).build();
+            allDescriptors.putIfAbsent(protoFileName, new CorfuRecord<>(protoFd, null));
```

Comment:
How can this generate duplicates if the condition in line 340 is 

```java

// If the fileDescriptorProto has already been added then continue.
            if (tableDescriptorsBuilder.getFileDescriptorsMap().containsKey(fileDescriptorProto.getName())) {
                continue;
            }
```

Reply:
That's correct, It should not generate duplicates. The de-duplication is really only across tables in the registry not within a given schema's set of protobufs.